docs/docsite/rst/porting_guides/porting_guide_base_2.11.rst

```diff
@@ -46,6 +46,15 @@ No notable changes
 Modules
 =======
 
+
+New Module Parameter Type
+-------------------------
+
+A new parameter type, ``str_strict``, is now available. This allows module authors to have Ansible take the action specified in ``ANSIBLE_STRING_CONVERSION_ACTION`` (``warn``, ``error``, or ``ignore`` and silently convert to a string) when a value in a ``str_strict`` field is not a string, such as unquoted integer or boolean values.
+
+This is a change from what was introduced in 2.8 where this behavior applied to *all* ``str`` type parameters.
```

Comment:
"introduced and later reverted" ?

Reply:
Yes, that's accurate. Turns out automatically casting to a string is a feature of `str` not a bug.

Reply:
I would say it's both a feature **and** a bug. Unfortunately it's so deeply integrated in the way Ansible works that it's really hard (or even impossible) to get rid of.

Reply:
Oh, I see... what a complex history this one has...... Hmmm.... since this was introduced in 2.8 which is now several years in the past, do we want to introduce str_strict or do we want to introduce str_relaxed?

* Argument for str_strict: 
  *  We can revert the behaviour for every other use of str.  That way maintainers don't have to do anything for people to stop seeing the warning in the common case.
* Argument for str_relaxed:
  * Anyplace where it truly matters that string conversion will cause problems will need to be updated otherwise they will suddenly lose the automatic checking they were counting on to stop users from doing the wrong thing.

I see major drawbacks to either of these cases.  So I'm just going to leave this comment for your evaluation.

Reply:
Judging by the issues raised since this change was made, most are opened due to the warning causing confusion and noise rather than identifying a useful problem.

I'm not aware of any cases where this brought to light a string conversion that would have silently caused a problem with the module execution. Doesn't mean there aren't any, though.

I'm in favor of preventing module maintainers from having to change their argument specs to get the pre-2.8 behavior of just casting `str` parameters to string.

Reply:
Well, I can just say that before the 2.8 change, there were several issues created for docker_container which were caused by unwanted string conversions. After the 2.8 change, I haven't seen any again (or at least right now I don't remember one). That's of course only anecdotical evidence, but I'm sure there was at least one case where this warning helped. Though probably less cases than where this was annoying and confusing :/

Reply:
That's a good data point. I'm glad it has helped somewhat. :)